orttraining/orttraining/core/graph/gradient_builder_base.cc

```diff
@@ -405,5 +405,22 @@ AttributeProto GradientBuilderBase::AttributeDefinitionToAttributeProto(
   return attr_proto;
 }
 
+// NOTE THAT we are changing the ATenOp forward node. If any of the input requires grad, we need to:
```

Comment:
Op's attribute is designed to hold **static** characteristic for a node's computation. 

I think we are breaking this assumption when manipulate the value of attribute for indicating which inputs needs gradient.

Reply:
let's discuss offline on finding another way to pass this info.